File: Script/translator/quickSort.cs
using System;

namespace EsQuickSort
{
    class Program
    {
        static void Main(string[] args)
        {
            int[] num = { 1, 5, 8, 12 };
            int[] min = new int[num.Length];
            int[] max = new int[num.Length];
            int i = 0;
            x(num, min, max, i);
        }

        static int x(int[] arr, int[] min, int[] max, int i)
        {
            if (arr == null || arr.Length==1)
            {
                return 0;
            }
            else
            {
                int p = arr[0];
                if (p >= arr[i]){
                    max[i] = arr[i];
                    i++;
                }
                else {
                    min[i] = arr[i];
                    i++;
                }
                return 1;
            }
        }
    }
}

File: Script/translator/quickSort.py
def quickSort(num: list[int], min_: list[int], max_: list[int], i: int) -> None:
    if (len(num) in {0, 1}):
        return 0;
    p: int = num[0]
    if (p >= num[i]):
        max_[i] = num[i]
        i += 1
    else:
        min_[i] = num[i]
        i += 1
    return 1;


if __name__ == '__main__':
    num = [1, 5, 12, 8]
    min_: list[int] = [0 for _ in range(len(num))]
    max_: list[int] = [0 for _ in range(len(num))]
    i: int = 0
    quickSort(num, min_, max_, i)
    print(num)